src/main/java/com/mraof/minestuck/item/weapon/OnHitEffect.java

```diff
@@ -145,6 +148,41 @@
 		target.attackEntityFrom(source, rng);
 	};
 	
+	OnHitEffect SHIELD_DISABLE = (stack, target, attacker) -> {
```

Comment:
As said elsewhere, forge seem to already have a hook for this accessible by implementing/overriding `canDisableShield()` in the item class. If that hook works as intended, I'd suggest you make use of that one instead, perhaps by adding a "disables shield" boolean in the weapon properties builder.